mm2src/mm2_db/src/indexed_db/db_lock.rs

```diff
@@ -14,30 +13,34 @@ pub struct ConstructibleDb<Db> {
     /// It's better to use something like [`Constructible`], but it doesn't provide a method to get the inner value by the mutable reference.
     mutex: AsyncMutex<Option<Db>>,
     db_namespace: DbNamespaceId,
-    wallet_rmd160: Option<H160>,
+    db_id: Option<String>,
 }
 
 impl<Db: DbInstance> ConstructibleDb<Db> {
     pub fn into_shared(self) -> SharedDb<Db> { Arc::new(self) }
 
     /// Creates a new uninitialized `Db` instance from other Iguana and/or HD accounts.
     /// This can be initialized later using [`ConstructibleDb::get_or_initialize`].
-    pub fn new(ctx: &MmArc) -> Self {
+    pub fn new(ctx: &MmArc, db_id: Option<&str>) -> Self {
+        let rmd = hex::encode(ctx.rmd160().as_slice());
+        let db_id = db_id.unwrap_or(&rmd);
         ConstructibleDb {
             mutex: AsyncMutex::new(None),
             db_namespace: ctx.db_namespace,
-            wallet_rmd160: Some(*ctx.rmd160()),
+            db_id: Some(db_id.to_string()),
```

Comment:
I initially believed it was only complicated for sqlite and it would be easier on the indexed db side, but it seems that it is just equally complicated problem...

Reply:
yeah but still not as sqlite lol 😄

Reply:
I don't plan to handle sql refactoring in this PR as it's kinda complex so I'll decide on the most efficient way to doing it in another PR as planned before. cc @onur-ozkan